crates/op-rbuilder/src/payload_builder_vanilla.rs

```diff
@@ -1355,3 +1410,88 @@ fn estimate_gas_for_builder_tx(input: Vec<u8>) -> u64 {
 
     zero_cost + nonzero_cost + 21_000
 }
+
+/// Extracts commitment from access list entries, pointing to 0x420..022 and validates them
+/// against supervisor.
+///
+/// If commitment present pre-interop tx rejected.
+///
+/// Returns (is_valid, is_recoverable)
+pub fn is_cross_tx_valid<N: NodePrimitives>(
+    tx: &N::SignedTx,
+    client: Option<&SupervisorValidator>,
```

Comment:
```suggestion
    client: &SupervisorValidator,
```
it seems like the caller is responsible if checking if interop is activated, so supervisor client isn't an optional param

Reply:
I want to simplify execute_best_transactions as much as possible, it would look consistent 
Don't the same for reth PR

Reply:
In the reth PR I will ensure that the supervisor is set during the building, but in here we need to rebase first

Reply:
makes sense